packages/camera/camera_android_camerax/lib/src/rotated_preview_utils.dart

```diff
@@ -0,0 +1,34 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+
+import 'camerax_library.dart' show Surface;
+
+/// Returns the number of counter-clockwise quarter turns represented by
+/// [surfaceRotationConstant], a [Surface] constant representing a clockwise
+/// rotation.
+int getQuarterTurnsFromSurfaceRotationConstant(int surfaceRotationConstant) {
+  return switch (surfaceRotationConstant) {
+    Surface.rotation0 => 0,
+    Surface.rotation90 => 3,
+    Surface.rotation180 => 2,
+    Surface.rotation270 => 1,
+    int() => throw ArgumentError(
+        '$surfaceRotationConstant is an unknown Surface rotation constant, so counter-clockwise quarter turns cannot be determined.'),
+  };
+}
+
+/// Returns the clockwise quarter turns applied by the CameraPreview widget
+/// based on [orientation], the current device orientation (see
+/// camera/camera/lib/src/camera_preview.dart).
```

Comment:
I'm slightly concerned that the rotation in this plugin depending on the rotation of the app-facing package. This comment doesn't block this PR, but in a future implementation of the entire `camera` package structure, each platform should probably handle its own rotations and the app facing shouldn't make any assumptions.

Reply:
Agreed. It does seem like Android is the only platform currently where rotation needs to be handled beyond the app-facing `CameraPreview` widget as of today, but nonetheless, I agree that organizationally, it should be handled within each platform as a result.